src/components/LogTable.tsx

```diff
@@ -1,46 +1,167 @@
 import { FirebaseContext } from '../data/FirebaseContext';
-import { useContext } from 'react';
+import { useContext, useState } from 'react';
 import { useLogs, deleteLog } from '../hooks/use-logs';
 import { ILog } from '../data/data-types';
 import { useInProgress } from '../hooks/use-in-progress';
+import { DateTime } from 'luxon';
 import Spinner from './Spinner';
+import { useAccount } from '../hooks/use-account';
 
 export function LogTable() {
   const fBaseContext = useContext(FirebaseContext)!;
-  const { logs } = useLogs(fBaseContext);
+
+  const [selectedList, setSelectedList] = useState<string | undefined>(undefined);
+
+  const { account: { recentList } } = useAccount(fBaseContext);
+  const { logs, lists } = useLogs(fBaseContext, selectedList);
+  logs.sort((a: ILog, b: ILog) => !a.endTime || !b.endTime ? 0 : b.endTime?.seconds - a.endTime?.seconds);
+
+  const groups = [];
+  let currentGroup = [];
+  let prevWeekNumber = 0;
+  for (const log of logs) {
+    const endDate = DateTime.fromSeconds(log.endTime!.seconds);
+    const thisWeekNumber = endDate.weekNumber;
+
+    if (prevWeekNumber !== thisWeekNumber) {
+      prevWeekNumber = thisWeekNumber;
+      if (currentGroup.length > 0) {
+        groups.push(currentGroup);
+        currentGroup = [];
+      }
+    }
+
+    currentGroup.push(log);
+  }
+
+  if (currentGroup.length > 0) {
+    groups.push(currentGroup);
+  }
 
   return (
+    <>
+    <div>
+      <select name='list-select' id='list-select' value={selectedList} onChange={e => setSelectedList(e.target.value)}>
+        <option>--</option>
```

Comment:
This only exists to work around the fact that filtering to a list removes all the other lists from the response that I use to determine what the lists are. 😆